File: PRG_321/week02/p717N10/p717N10/SingleFamily.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace p717N10
{
    public class SingleFamily : Housing, IUnits
    {
        private double m_SquareFootage;
        private bool m_GarageAvailable;

        public double SquareFootage
        {
            set { m_SquareFootage = value;  }
            get { return m_SquareFootage; }
        }

        public override string ToString()
        {
            String old_content =  base.ToString();
            String new_Content = " - Sqft = " + m_SquareFootage + " - garage = ";
            if( m_GarageAvailable == true )
            {
                new_Content += " available";
            } else
            {
                new_Content += " NOT available";
            }
            return old_content + new_Content;
        }
        public bool GarageAvailable
        {
            set { m_GarageAvailable = value;  }
            get { return m_GarageAvailable;  }
        }
    }
}

File: PRG_351/Java_Mod_5/src/java_assignments/SingleFamily.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package java_assignments;

/**
 *
 * @author Esposito
 */
public class SingleFamily
{

        private double m_SquareFootage;
        private boolean m_GarageAvailable;

        
        public double GetSquareFootage()
        {
            return m_SquareFootage;
        }
        
        public void SetSquareFootage(double v)
        {
            m_SquareFootage = v;
        }


        public String ToString()
        {
            String new_Content = " - Sqft = " + m_SquareFootage + " - garage = ";
            if( m_GarageAvailable == true )
            {
                new_Content += " available";
            } else
            {
                new_Content += " NOT available";
            }
            return new_Content;
        }
        
        public void SetGarageAvailable( boolean input )
        {
            m_GarageAvailable = input;
        }
        
        public boolean GetGarageAvailable( )
        {
            return m_GarageAvailable;
        }

    }  

